src/components/pages/dashboard/dashboard.js

```diff
@@ -66,38 +72,41 @@ export class Dashboard extends Component {
     // Telemetry stream - START
     const onPendingStart = () => this.setState({ telemetryIsPending: true });
 
-    const telemetry$ = TelemetryService.getTelemetryByDeviceIdP15M()
+    const getTelemetryStream = ({ deviceIds = [] }) => TelemetryService.getTelemetryByDeviceIdP15M(deviceIds)
       .merge(
         this.telemetryRefresh$ // Previous request complete
           .delay(Config.telemetryRefreshInterval) // Wait to refresh
           .do(onPendingStart)
-          .flatMap(_ => TelemetryService.getTelemetryByDeviceIdP1M())
+          .flatMap(_ => TelemetryService.getTelemetryByDeviceIdP1M(deviceIds))
       )
       .flatMap(transformTelemetryResponse(() => this.state.telemetry))
       .map(telemetry => ({ telemetry, telemetryIsPending: false })); // Stream emits new state
       // Telemetry stream - END
 
       // KPI stream - START
-      const currentFrom = 'NOW-PT1H';
-      const previousFrom = 'NOW-PT2H';
-
-      const currentParams = { from: currentFrom, to: 'NOW' };
-      const previousParams = { from: previousFrom, to: currentFrom };
 
       // TODO: Add device ids to params - START
-      const kpis$ = this.panelsRefresh$
+      const getKpiStream = ({ deviceIds = [], timeInterval }) => this.panelsRefresh$
         .delay(Config.dashboardRefreshInterval)
         .startWith(0)
         .do(_ => this.setState({ kpisIsPending: true }))
-        .flatMap(_ =>
-          Observable.forkJoin(
+        .flatMap(_ => {
+
+          const devices = deviceIds.length ? deviceIds.join(',') : undefined;
```

Comment:
nit: Extra lines in between. I don't we are consistent with where we leave extra lines and where we don't. This segment some extra lines in between.

Reply:
I agree and was actually thinking the same thing earlier when I was looking at this code again. I'll update the newlines here in a later PR.